torch_xla/core/dynamo_bridge.py

```diff
@@ -464,6 +488,9 @@ def extract_compiled_graph(xla_model: torch.fx.GraphModule, xla_args):
   # This call is critical to make sure xla_args' tensor id show up in graph_input_tensor_ids
   xm.mark_step()
 
+  # Move tensor constructor nodes that do not target XLA devices.
```

Comment:
I am a bit confuse by `move`, where does it move to? Does it auto convert the cpu tensor to xla tensor?

Reply:
It replaces `device=cpu` by `device=xla` (1) in the tensor constructor nodes it finds (2).

(1) uses an XLA device being used in the program
(2) checks whether it's "movable", first.

A tensor constructor node is movable iff all of its uses (recursively) don't reach an un-movable node (see [`cannot_be_moved` function](https://github.com/pytorch/pytorch/blob/5de0dff7ea2f0857f2682ce4c1e6ad70779a76a6/torch/_inductor/fx_passes/post_grad.py#L1027)).